src/lib.rs

```diff
@@ -9,44 +9,50 @@
 //! # fn main() {
 //! // We can parse a string representation of SK combinatory logic,
 //! let sk_rules = "S x_ y_ z_ = (x_ z_) (y_ z_); K x_ y_ = x_;";
-//! let parsed_trs = parse_trs(&vec![], sk_rules).expect("parsed TRS");
+//! let (_sig, trs) = parse_trs(vec![], sk_rules).expect("parsed TRS");
 //!
 //! // and we can also parse an arbitrary term.
 //! let term = "S K K (K S K)";
-//! let parsed_term = parse_term(&vec![], term).expect("parsed term");
+//! let (_sig, parsed_term) = parse_term(vec![], term).expect("parsed term");
 //!
 //! // These can also be constructed by hand. Let's look at the term:
-//! let app = Op::new(0, 2, Some(".".to_string()));
-//! let s = Op::new(1, 0, Some("S".to_string()));
-//! let k = Op::new(2, 0, Some("K".to_string()));
+//! let symbols = vec![
+//!     (2, Some(".".to_string())),
+//!     (0, Some("S".to_string())),
+//!     (0, Some("K".to_string())),
+//! ];
+//! let (sig, operators) = Signature::new(symbols);
+//! let app = operators[0];
+//! let s = operators[1];
+//! let k = operators[2];
```

Comment:
Just looking at the diff here, this new code is much more verbose. Some sort of `find` operation in the `Signature` might help us make this more compact, or perhaps just using `new_op` and `new_var` here instead.